Document zenodo deposits (#1547)
openjournals/joss: docs/submitting.md

- **JOSS reviews are iterative and conversational in nature.** Reviewers are encouraged to post comments/questions/suggestions in the review thread as they arise, and authors are expected to respond in a timely fashion.
- **Authors are expected to respond to reviewer feedback promptly.** We generally ask that authors respond to reviewer comments and questions within 2 weeks, and complete requested changes within 4-6 weeks, matching the commitment we ask of reviewers, unless otherwise negotiated with the editor. If you need more time (for example, if the changes requested are particularly substantial), please communicate this to the reviewers and editor on the review thread. Prolonged unresponsiveness may result in the paper being rejected due to lack of engagement.
- Authors and reviewers are asked to be patient when waiting for a response from an editor. Please allow a week for an editor to respond to a question before prompting them for further action.

> If you want to learn more details about the review process, take a look at the [reviewer guidelines](reviewer_guidelines).

Reviewer: ```suggestion
Further details about the review process are available in the [reviewer guidelines](reviewer_guidelines).
```

## Post-Review

> - Upon successful completion of the review, authors will make a tagged release of the software. Releases must include the complete contents of the repository at the tagged release commit. It is recommended to also include built artifacts like compiled binaries, when applicable, reasonable, and supported by the archive.

Author: i should also flag this as new language - this is an archivist's definition of a release, which might be different than a software developer's. some repos might only create releases with some reduced source distribution which is definitely appropriate for language-specific packaging needs. the idea here was that for a publication archive we have different needs, and that is preserving the state of the work as completely as possible at the time it was evaluated. so idk if this needs clearer signposting or if we want to not have this language, but realize it might fly under the rader unless pointed to

Author: the reason i added it is that some submissions might not have a standard release process so then this requirement is pretty ambiguous - the release could be empty, in one extreme case, and this could look very different across applications and languages. so give a clear brightline: "archive the repo"

Author: this language has moved, new comment thread: https://github.com/openjournals/joss/pull/1547/changes#r3284732669

> - Upon successful completion of the review, authors will make a tagged release of the software. Releases must include the complete contents of the repository at the tagged release commit. It is recommended to also include built artifacts like compiled binaries, when applicable, reasonable, and supported by the archive.

Reviewer 2: I'm very uncertain about "It is recommended to also include built artifacts like compiled binaries, when applicable, reasonable, and supported by the archive."

Author: So the idea is as an archival copy, baseline is the complete source, but e.g. if in the future it becomes impossible to compile the software, like say you have a rust crate and it has dependencies that are yanked and no longer available, then it would be good to have archived copies of the binary. I figure this is more of an "extra" step that the authors could decide what is reasonable to archive, because we wouldn't want people to e.g. be archiving huge docker containers or whatnot. Most normal releases would include compiled binaries, when appropriate, anyway.

Reviewer 2: I guess the "It is recommended" part is what I was most uncertain about.  Maybe this could be "Consider including ..."?

Author: Oh gotcha, yes of course.

Author: done in https://github.com/openjournals/joss/pull/1547/commits/309384d42af963151177c270ea99337473fb0df5

> - Authors will deposit a copy of the repository at the tagged release with a data-archiving service such as [Zenodo](https://zenodo.org/) (see [below](#archiving-with-zenodo)) or [figshare](https://figshare.com/), get a DOI for the archive, and update the review issue thread with the version number and DOI.

Reviewer 2: ```suggestion
- Authors will deposit a copy of the repository at the tagged release with a research data repository, typically one that is listed in [re3data.org](https://re3data.org), such as [Zenodo](https://zenodo.org/) (see [below](#archiving-with-zenodo)), [figshare](https://figshare.com/), or an institutional repository. The repository will accept the release and its associated metadata, and will provide a DOI for the deposit. The Author then updates the review issue thread with the version number of the release and the DOI from the data research data repository.
```

- After we assign a DOI for your accepted JOSS paper, its metadata is deposited with CrossRef and listed on the JOSS website.
- The review issue will be closed, and an automatic post from [@JOSS at Mastodon](https://fosstodon.org/@joss) will announce it!

Optionally:
- Authors may wish to add a `CITATION.cff` file referencing the JOSS publication. 
  See the [`CITATION.cff` docs](https://citation-file-format.github.io/) or 
  [Zenodo's citation docs](https://help.zenodo.org/docs/github/describe-software/citation-file/).

### Archiving with Zenodo

Repository archives at Zenodo can either be made through Zenodo's automatic release depositing feature

> or manually. 

Reviewer 2: ```suggestion
or manually. (Note that the manual process described below is basically what can be done with any research data repository, not just Zenodo.)
```

Author: Might be a bit cluttery, this is a section on "how to deposit with zenodo" rather than "how to deposit, period," if we want to have several different guides for different archives we can split that off to a separate page maybe?

Reviewer 2: Maybe we can separate the manual deposit instructions, which aren't really zenodo-specific, from the automatic instructions, which is very zenodo-specific

Author: Do you mean something like having some first section that is "Requirements for deposits in general" and then followed by the zenodo specific instructions? I don't think I understand your meaning but would be happy to restructure however. 

The instructions on manual deposits for zenodo are just for completeness sake - intended decision flow is supposed to read like "first choose archive. If not zenodo, then {not documented}. If zenodo, choose manual or automatic," but if it's reading like "there are two choices, both zenodo, manual or automatic" then that's a problem.

Reviewer 2: I read it as
 * here are instructions on how to use zenodo
   * automatically
   * manually

My suggestion is
 * here are instructions
   * automatic on zenodo
   * manual on zenodo or another place

Author: Alright, can do that, and can generalize the language a bit then. One moment

Author: OK I did that here: https://github.com/openjournals/joss/pull/1547/commits/309384d42af963151177c270ea99337473fb0df5

Reorganizing it that way did help clarity a bit i think - all the details about what the archival release should include are split out as their own set of requirements (and those can be changed, [separate comment thread](https://github.com/openjournals/joss/pull/1547/changes#r3284732669)) rather than being stuffed in the post-review checklist bullet points. Tried to make the manual archiving steps generic, they're maybe not as helpful for manual archives at zenodo anymore, but they probably better communicate that "other repositories are allowed." How does that look?

Reviewer 2: thanks

## Post-Review

- Upon successful completion of the review, authors will make a tagged release of the software. Releases must include the complete contents of the repository at the tagged release commit. It is recommended to also include built artifacts like compiled binaries, when applicable, reasonable, and supported by the archive.
- Authors will deposit a copy of the repository at the tagged release with a data-archiving service such as [Zenodo](https://zenodo.org/) (see [below](#archiving-with-zenodo)) or [figshare](https://figshare.com/), get a DOI for the archive, and update the review issue thread with the version number and DOI.
- After we assign a DOI for your accepted JOSS paper, its metadata is deposited with CrossRef and listed on the JOSS website.
- The review issue will be closed, and an automatic post from [@JOSS at Mastodon](https://fosstodon.org/@joss) will announce it!

Optionally:
- Authors may wish to add a `CITATION.cff` file referencing the JOSS publication. 
  See the [`CITATION.cff` docs](https://citation-file-format.github.io/) or 
  [Zenodo's citation docs](https://help.zenodo.org/docs/github/describe-software/citation-file/).

> 

Author: I can take this out if we don't actually recommend this, but afaik this is the standard programmatic way to declare citation information in repositories, which authors may want to do/if they haven't heard about this it might be helpful to tell them

Reviewer 3: `CITATION.cff` is way more understandable than `zenodo.json` and I think what we should be encouraging in general. 👍

Author: also good point, we don't have to be docs for what zenodo supports and should probably make a clearer recommendation / be concise https://github.com/openjournals/joss/pull/1547/commits/5363c4f714bc0b040aa0b0d81fe180334ca1dd8a

Reviewer: One or the other is good, but note that you can't currently have both: https://github.com/citation-file-format/citation-file-format/issues/374

Author: I can add that in a footnote or something? edit: seems like that might just be noisy, reading that issue. i think if we just recommend using `CITATION.cff` that covers it, since the citation file overrides the zenodo file anyway.

Reviewer 2: I think this is fine

### Archiving with Zenodo

Repository archives at Zenodo can either be made manually, 
or through Zenodo's automatic release depositing feature. 
It is recommended to create either a `CITATION.cff` or a 
[`.zenodo.json`](https://help.zenodo.org/docs/github/describe-software/zenodo-json/) file so the JOSS metadata is incorporated in the Zenodo deposit.

#### Manual Archiving

See: 
[Upload software manually](https://help.zenodo.org/docs/github/archive-software/manual-upload/) and 
[Create new upload](https://help.zenodo.org/docs/deposit/create-new-upload)

- Download the files from the GitHub release
- Create a new [Zenodo upload](https://help.zenodo.org/docs/deposit/create-new-upload/)
- Upload the zipped files from the release.
  When depositing manually, you must upload a single .zip file.
- Select "Software" as the resource type, 
  and add [metadata](https://help.zenodo.org/docs/deposit/describe-records/) for the deposit
- Publish the record

> #### Automatic Archiving

Reviewer 3: Maybe flip these so the more maintainable way (the Zenodo integration) comes first?

Author: agreed, done https://github.com/openjournals/joss/pull/1547/commits/b754658a60eff729ff830912d81e798f2fc1f0d3

- **JOSS reviews are iterative and conversational in nature.** Reviewers are encouraged to post comments/questions/suggestions in the review thread as they arise, and authors are expected to respond in a timely fashion.
- **Authors are expected to respond to reviewer feedback promptly.** We generally ask that authors respond to reviewer comments and questions within 2 weeks, and complete requested changes within 4-6 weeks, matching the commitment we ask of reviewers, unless otherwise negotiated with the editor. If you need more time (for example, if the changes requested are particularly substantial), please communicate this to the reviewers and editor on the review thread. Prolonged unresponsiveness may result in the paper being rejected due to lack of engagement.
- Authors and reviewers are asked to be patient when waiting for a response from an editor. Please allow a week for an editor to respond to a question before prompting them for further action.

Further details about the review process are available in the [reviewer guidelines](reviewer_guidelines).

## Post-Review

> Upon successful completion of the review:

Reviewer 2: ```suggestion
Upon successful completion of the review, the editor will ask the submitter to do the following:
```

> - Authors will make a tagged, archival release of the software in the GitHub[^otherforges] repository (see [below](#archiving-the-submission)).

Reviewer 2: ```suggestion
- The submitter will make a tagged, archival release of the software in the GitHub or other[^otherforges] repository(ies) (see [below](#archiving-the-submission)).
```

> - Authors will deposit the archival release with a research data repository. The repository will accept the release and its associated metadata, and will provide a DOI for the deposit. 

Reviewer 2: ```suggestion
- The submitter will deposit the archival release in a research data repository. The repository will accept the release and its associated metadata, and will provide a DOI for the deposit. 
```

> - The Author then updates the review issue thread with the version number of the release and the DOI from the data research data repository.

Reviewer 2: ```suggestion
- The submitter will then update the review issue thread to provide the version number of the release and the DOI from the data research data repository to the editor.
```

- After we assign a DOI for your accepted JOSS paper, its metadata is deposited with CrossRef and listed on the JOSS website.
- The review issue will be closed, and an automatic post from [@JOSS at Mastodon](https://fosstodon.org/@joss) will announce it!

Optionally:

> - Authors may wish to add a `CITATION.cff` file referencing the JOSS publication. 

Reviewer 2: ```suggestion
- The submitter may wish to add a `CITATION.cff` file referencing the JOSS publication. 
```

  See the [`CITATION.cff` docs](https://citation-file-format.github.io/) or 
  [Zenodo's citation docs](https://help.zenodo.org/docs/github/describe-software/citation-file/).

[^otherforges]: Repositories hosted on code forges other than GitHub should use an analogous "release" feature, if available.

### Archiving the Submission

> Submissions must be archived in a repository to ensure long-term availability of the work.

Reviewer 2: ```suggestion
The software that was submitted and reviewed must be archived in a repository to ensure long-term availability of the work.
```

> Repositories are typically those listed in [re3data.org](https://re3data.org), 

Reviewer 2: ```suggestion
Archival repositories are typically those listed in [re3data.org](https://re3data.org), 
```

such as [Zenodo](https://zenodo.org/), 
[figshare](https://figshare.com/), 
or an institutional repository. 

Archival releases **must:**
- Correspond to a git tag with the current software version[^archivetag] in the submitted repository or repositories.

> - Include the complete contents of material considered during review. Typically this will be the contents of the repository, but in the case of monorepo submissions that include unreviewed material, the unreviewed material may be omitted.

Author: I loosened up this language a bit because we do have some submissions from monorepos or repos with other "test" or "demo" code in it, i'm not sure if this helps or hurts clarity so feedback plz. 

Same sort of trickiness applies as before - the needs of an archival release are a bit different than a typical "source distribution" release that repositories might already be set up to make, which usually only include the source of the package and not the docs and tests and other parts of the submission which may be important to evaluate a submission retroactively. 

I am picturing the situation where maybe in 20 years someone is looking at a JOSS submission long after the repository itself may have been deleted - what would someone need to be able to evaluate, understand, and ideally use the submission? So I am not sure if we need more or less scaffolding for this, or if other people think this is overkill and source distribution deposits are fine.

Reviewer 2: Some submissions also exist in multiple repos

Author: right, yeah, i was hoping the "material considered during review" and previous line saying "submitted repository or repositories" covered that, but i can made that more explicit in the second line if that's needed.

Reviewer 2: I guess it's ok as is

Author: no problem to change if it needs to be clearer! just lmk

Reviewer 2: I've suggested one small change, in the process of trying to read this whole thing again

> - Include the complete contents of material considered during review. Typically this will be the contents of the repository, but in the case of monorepo submissions that include unreviewed material, the unreviewed material may be omitted.

Reviewer 2: ```suggestion
- Include the complete contents of material considered during review. Typically this will be the contents of the repository(ies), but in the case of monorepo submissions that include unreviewed material, the unreviewed material may be omitted.
```

> Archival releases **may** optionally, at the Author's discretion:

Reviewer 2: ```suggestion
Archival releases **may** optionally, at the submitter's discretion:
```
